pyrsistent/typing.pyi

```diff
@@ -23,26 +23,34 @@ from typing import Union
 from typing import overload
 
 T = TypeVar('T')
+U = TypeVar('U')
 KT = TypeVar('KT')
 VT = TypeVar('VT')
 
+T_PMap = TypeVar('T_PMap', bound='PMap')
```

Comment:
I'm not an expert on Python type hints, what do these aliases buy us in comparison to the existing declarations?

Reply:
these TypeVars are the backwards-compatible way to achieve a `Self` type (see https://peps.python.org/pep-0673/ ) such that the type checker knows that a method defined in a base class can return a more specific subclass. This is relevant to preserve the `Checked*` types without having to re-define all methods from the base unchecked types, as well as for `PClass` which inherits from `PMap`, and even more importantly, for any user-defined subclass of `PClass`